src/Bicep.Core/Syntax/DecoratorSyntax.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class DecoratorSyntax : SyntaxBase
+    {
+        public DecoratorSyntax(Token at, SyntaxBase expression)
+        {
+            AssertTokenType(at, nameof(at), TokenType.At);
+            AssertSyntaxType(
+                expression,
+                nameof(expression),
+                typeof(SkippedTriviaSyntax),
+                typeof(VariableAccessSyntax),
+                typeof(PropertyAccessSyntax),
+                typeof(FunctionCallSyntax),
+                typeof(InstanceFunctionCallSyntax));
+
+            this.At = at;
+            this.Expression = expression;
+        }
+
+        public Token At { get; }
+
+        public SyntaxBase Expression { get; }
+
+        public IEnumerable<FunctionArgumentSyntax> Arguments => this.Expression switch
+        {
+            FunctionCallSyntax functionCall => functionCall.Arguments,

```

Comment:
>FunctionCallSyntax [](start = 12, length = 18)

I added a common base class to simplify stuff like this btw.

Reply:
Could you point me to the location of the class? I tried to search for it but didn't find it.

Reply:
It's the `FunctionCallSyntaxBase` class. Both `FunctionCallSyntax` and `InstanceFunctionCallSyntax` are derived from it.

Reply:
I don't mind if it comes on another PR (if at all).

Reply:
Gotcha. I was searching for wrong thing...I thought you meant simplifying extracting arguments. I updated it to use `FunctionCallSyntaxBase` since the change is simple.